lib/async/http/faraday/adapter.rb

```diff
@@ -32,8 +32,8 @@ def call(env)
 					client = HTTP::Client.new(*endpoints_for(env).to_a)
 					
 					response = client.send(env[:method], env[:url].request_uri, env[:request_headers], env[:body] || [])
-					
-					save_response(env, response.status, response.body.read, response.headers)
+
+					save_response(env, response.status, response.body && response.body.join, response.headers)
```

Comment:
`response.body && response.body.join` can be achieved more simply by `response.read`

https://github.com/socketry/protocol-http/blob/master/lib/protocol/http/body/reader.rb#L37-L44

Can you update the PR and I will merge.

Reply:
The reason why this bug exists is because this code was written before that method existed :)

Reply:
fixed, thanks!